### PR TITLE
Show roaming walker paths

### PR DESCRIPTION
**Pull request by crudelios**
- Show them on the construction ghost and building info
- It is an UI option which needs to be enabled first
- This replaces the "show walker waypoints" config as it is much more complete

This also allows you to easily check for proper forced walker mechanics when placing the building.

Possible improvement in the future: show all paths travelled by roamer type on corresponding overlay

### CODE REVIEW

**Comment by PrettyFlower** [comment]
This is a really cool feature! Tested it a bit and it seems to work, too. Just a couple of minor questions.

**Comment by crudelios** [comment]
Thanks!

Sure, but that will only work for buildings which don't spawn roamers. It should also only work when global labour pool is disabled.

**Comment by crudelios** [comment]
@bvschaik If you want to, this can easily be backported to Julius 🙂

**Review by PrettyFlower — COMMENTED** [review]
Final review comment 2

File: src/figure/figure.h [hunk 1]
```diff
@@ -22,7 +24,7 @@ typedef struct {
     unsigned char use_cross_country;
     unsigned char is_friendly;
     unsigned char state;
-    unsigned char faction_id; // 1 = city, 0 = enemy
+    unsigned char faction_id; // 2 = ghost, 1 = city, 0 = enemy
```
[the remark this answers is not in the document]
Comment (crudelios) [reply src/figure/figure.h:27]: Ah, good catch. I originally called the roamer preview a ghost early in the code, but that is now outdated.

File: src/widget/city_with_overlay.c [hunk 2]
```diff
@@ -352,6 +353,24 @@ void city_with_overlay_draw_building_footprint(int x, int y, int grid_offset, in
     }
 }
 
+static void draw_roamer_frequency(int x, int y, int grid_offset)
```
[the remark this answers is not in the document]
Comment (crudelios) [reply src/widget/city_with_overlay.c:356]: Yeah me too, but both codes are duplicated between `city_with_overlay` and `city_without_overlay`.

What I want to do in the future is to merge both files, but while that doesn't happen I'm not very worried about duplicating this code.

**Pull request merged**